spec/_plugins/locale_url_filter_spec.rb

```diff
@@ -0,0 +1,278 @@
+require 'jekyll'
+require './_plugins/locale_url_filter.rb'
+
+class JekyllFilter
+  include Jekyll::LocaleUrlFilter
+
+  def initialize(config, page_data)
+    @site = Jekyll::Site.new(Jekyll::Configuration.from(config))
+    page = Jekyll::Page.new(@site, @site.source, '', 'README.md')
+    page.data.merge!(page_data)
+    @context = Liquid::Context.new(@site.site_payload, {}, site: @site, page: page)
+  end
+end
+
+RSpec.describe Jekyll::LocaleUrlFilter do
+  let(:config) { {} }
+  let(:page_data) { { 'lang' => 'en' } }
+  let(:instance) { JekyllFilter.new(config, page_data) }
+
+  describe '#locale_url' do
+    let(:path) { '' }
+    let(:locale) { nil }
+    subject { instance.locale_url(path, locale) }
```

Comment:
nit, not required, personally I really like named subjects because they make it easier for me to remember what we're testing

```suggestion
    subject(:locale_url) { instance.locale_url(path, locale) }
```
then we can use `locale_url` instead of `subject` in the specs

Reply:
> nit, not required, personally I really like named subjects because they make it easier for me to remember what we're testing

TIL! Updated in 07959a3.